### PR TITLE
IdP name configuration

### PR DESCRIPTION
**Pull request by tinakurian**
making IdP configurable via configmap

Jira: https://issues.redhat.com/browse/CRT-743
Relates to: codeready-toolchain/toolchain-common#119

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -26,6 +26,7 @@ require (
 
 replace (
 	github.com/Azure/go-autorest => github.com/Azure/go-autorest v13.0.0+incompatible
+	github.com/codeready-toolchain/toolchain-common => github.com/tinakurian/toolchain-common v0.0.0-20200727140128-cfe3823faae8
```
Comment (tinakurian) [inline go.mod:29]: reminder to myself!

File: deploy/env/e2e-tests.yaml [hunk 2]
```diff
@@ -1,3 +1,2 @@
 member-operator:
-  identity-provider: 'testIdP'
```
Comment (MatousJobanek) [inline deploy/env/e2e-tests.yaml:2]: Correct me if I'm wrong, but you should not remove these values from the env config files. This is btw the reason why your e2e tests are failing.
I understand that you need to primarily use the value from the ConfigMap, but you can still combine them and replace it with the value from CM, right?
Or, you could rename it to `default-identity-provider` and use this value as the default one if no value is specified in the CM.

File: pkg/controller/useraccount/useraccount_controller_test.go [hunk 3]
```diff
@@ -33,7 +33,9 @@ func TestReconcile(t *testing.T) {
 	logf.SetLogger(logf.ZapLogger(true))
 	username := "johnsmith"
 	userID := uuid.NewV4().String()
-	config := configuration.LoadConfig()
+	cfg, err := configuration.LoadConfig(test.NewFakeClient(t))
+	require.NoError(t, err)
+	config := cfg
```
Comment (MatousJobanek) [inline pkg/controller/useraccount/useraccount_controller_test.go:38]: could you please tell me why you reassign the config variable? Wouldn't it be shorter to just assign it directly to `config`?:
```suggestion
	config, err := configuration.LoadConfig(test.NewFakeClient(t))
	require.NoError(t, err)
```

File: pkg/controller/useraccount/useraccount_controller_test.go [hunk 4]
```diff
@@ -841,7 +843,8 @@ func TestDisabledUserAccount(t *testing.T) {
 	s := scheme.Scheme
 	err := apis.AddToScheme(s)
 	require.NoError(t, err)
-	config := configuration.LoadConfig()
+	cfg, err := configuration.LoadConfig(test.NewFakeClient(t))
+	config := cfg
```
Comment (MatousJobanek) [inline pkg/controller/useraccount/useraccount_controller_test.go:847]: same here, plus you forgot to check the error:
```suggestion
	config, err := configuration.LoadConfig(test.NewFakeClient(t))
	require.NoError(t, err)
```

File: pkg/controller/useraccount/useraccount_controller_test.go [hunk 5]
```diff
@@ -1211,7 +1214,9 @@ func prepareReconcile(t *testing.T, username string, initObjs ...runtime.Object)
 	err := apis.AddToScheme(s)
 	require.NoError(t, err)
 	fakeClient := test.NewFakeClient(t, initObjs...)
-	config := configuration.LoadConfig()
+	cfg, err := configuration.LoadConfig(fakeClient)
+	require.NoError(t, err)
+	config := cfg
```
Comment (MatousJobanek) [inline pkg/controller/useraccount/useraccount_controller_test.go:1219]: same here 
```suggestion
	config, err := configuration.LoadConfig(fakeClient)
	require.NoError(t, err)
```

File: pkg/configuration/configuration_test.go [hunk 6]
```diff
@@ -47,6 +51,64 @@ func TestGetAllMemberParameters(t *testing.T) {
...
+	t.Run("default", func(t *testing.T) {
+		// when
+		config := getDefaultConfiguration(t)
+
+		// then
+		assert.Equal(t, "rhd", config.GetIdP())
+	})
+	t.Run("env overwrite", func(t *testing.T) {
+		// given
+		restore := test.SetEnvVarAndRestore(t, "MEMBER_OPERATOR_CONFIG_MAP_NAME", "test-config")
```
Comment (MatousJobanek) [inline pkg/configuration/configuration_test.go:66]: we should set and restore also the variables retrieved from the ConfigMap:
```suggestion
		restore := test.SetEnvVarsAndRestore(t,
			test.Env("MEMBER_OPERATOR_CONFIG_MAP_NAME", "test-config"),
			test.Env("MEMBER_OPERATOR_IDENTITY_PROVIDER", ""),
			test.Env("MEMBER_OPERATOR_TEST_TEST", ""))
```

File: deploy/env/e2e-tests.yaml (hunk 2 above)
Comment (tinakurian) [reply deploy/env/e2e-tests.yaml:2]: @MatousJobanek shouldn't the default be set via viper? the same way we do for other values that are being set via the configMap?

File: pkg/controller/useraccount/useraccount_controller_test.go (hunk 3 above)
Comment (tinakurian) [reply pkg/controller/useraccount/useraccount_controller_test.go:38]: yeah, definitely didn't mean to do that, must have been a bit tired when i did this lol, will fix

Comment (tinakurian) [reply pkg/controller/useraccount/useraccount_controller_test.go]: fixed here: a7cfb415bcaa14a4fe90ea746eb7dfc59185eb65

File: pkg/controller/useraccount/useraccount_controller_test.go (hunk 4 above)
Comment (tinakurian) [reply pkg/controller/useraccount/useraccount_controller_test.go]: fixed here: a7cfb415bcaa14a4fe90ea746eb7dfc59185eb65 and here: c5066c27bced708ae2f020d83113d041fbe39690

File: pkg/controller/useraccount/useraccount_controller_test.go (hunk 5 above)
Comment (tinakurian) [reply pkg/controller/useraccount/useraccount_controller_test.go]: fixed here: a7cfb415bcaa14a4fe90ea746eb7dfc59185eb65

File: pkg/configuration/configuration_test.go (hunk 6 above)
Comment (tinakurian) [reply pkg/configuration/configuration_test.go]: fixed here: 8a2e7be9cb97a69f7afb5490c1ab47e251946fda

File: deploy/env/e2e-tests.yaml (hunk 2 above)
Comment (MatousJobanek) [reply deploy/env/e2e-tests.yaml:2]: It's up to us how we deal with the default values - if we use viper functionality or some other way...
I see that you changed the IdP name to `rhd` for the e2e tests - this is a feasible solution as well and much easier :+1:

File: pkg/configuration/configuration.go [hunk 7]
```diff
@@ -50,8 +48,13 @@ const (
 
 	ClusterUnavailableDelay        = "cluster.unavailable.delay"
 	DefaultClusterUnavailableDelay = "60s"
+
+	identityProviderName        = "identity.provider"
+	DefaultIdentityProviderName = "rhd"
```
Comment (MatousJobanek) [inline pkg/configuration/configuration.go:53]: it's just a detail - not sure why you move it - you put the constants to the part containing KubeFed configuration constants

File: pkg/configuration/configuration.go [hunk 8]
```diff
@@ -50,8 +48,13 @@ const (
...
 )
 
+var log = logf.Log.WithName("configuration")
```
Comment (MatousJobanek) [inline pkg/configuration/configuration.go:56]: not used 
```suggestion
```

File: pkg/configuration/configuration.go (hunk 7 above)
Comment (tinakurian) [reply pkg/configuration/configuration.go]: fixed here: cecef2d2b60d85e6e54793c4d68db1168ee8c449

File: pkg/configuration/configuration.go (hunk 8 above)
Comment (tinakurian) [reply pkg/configuration/configuration.go]: fixed here: cecef2d2b60d85e6e54793c4d68db1168ee8c449

File: pkg/configuration/configuration.go (hunk 7 above)
Comment (tinakurian) [reply pkg/configuration/configuration.go]: @MatousJobanek i just noticed that all the kubefed vars are exported - is there any reason for this? i just checked and it doesn't look like they are used anywhere other than in configuration. if there is no reason, i can fix this in this PR

Comment (MatousJobanek) [reply pkg/configuration/configuration.go]: You can keep it there - I'm going to remove most of the constants in the next PRs anyway.

**Pull request merged**